### PR TITLE
fix: auth flow works, storing email and adress optionally

### PR DESCRIPTION
**Pull request by ponikar**
Refactoring `secrets.ts`
- Update `API_END_POINT`, `AUTH_BASE_URL` and `HUDDLE_THIRD_PARTY_API_KEY` while moving to production. 
- Removed token refreshing logic in `service.ts`, since we have lifetime token validity. 
- Removed `HUDDLE_API_KEY`since we have new flow. 

New Flow 
- redirect user to new dashboard and generate identity token after successfully login.
- store identity token and pass `HUDDLE_THIRD_PARTY_API_KEY` and `identityToken` while making an API call to platform server. 

Improvements 
- removed `subdomainId` since name is enough to validate and create meeting. refactored logic. `subdomain` prop will make sure to create a meeting, `app` is default subdomain. 
- removed `access_token` and `refresh_token` logic
- Rendering home screen on `onGmailCompose`
- Clean up `Service.ts` file and removed unnecessary token refreshing logic.

### CODE REVIEW

**Comment by arcinston** [comment]
@ponikar one QOF improvement could be adding biome to this repo
you can use the config file biome.json from shinigami repo

there is a lot of var usage inherently in the code , when it was written 
now we can clean it up , add linting etc 

whats your opinion? @ponikar

File: src/utils.ts [hunk 1]
```diff
@@ -49,51 +49,67 @@ function handleSwitchChange() {
...
+    },
+    muteHttpExceptions: true,
+  });
+
+  const data = JSON.parse(response.getContentText()) as {
+    subdomains: {
+      id: string;
+      url: string;
+      name: string;
+      createdAt: string;
```
Comment (arcinston) [inline src/utils.ts:105]: all we need for calenders is the name of the subdomain right?

rest info contains sensitive info like apikey projectid which should not be returned

@ponikar can discuss which fields are critical for the functioning of the calender func and future addons we might create on top of the apis

File: src/pages/addMeeting.ts [hunk 2]
```diff
@@ -97,19 +94,26 @@ const createAddMeetingCardSection = (subject: string) => {
...
-      } else {
-        subdomainSwitch.addItem(subdomain.name, subdomain.id, true);
+      .setFieldName("subdomain");
+
+    subdomainResponse.subdomains.push({
+      id: "app",
+      name: "app",
+      url: "",
+      createdAt: "",
+      updatedAt: "",
```
Comment (arcinston) [inline src/pages/addMeeting.ts:104]: why store all this data (i.e url , createdAt , etc etc)

also the fetch subdomains query should not return this data if it is being used for calenders purposes

File: src/pages/addMeeting.ts [hunk 3]
```diff
@@ -49,40 +54,32 @@ const createAddMeetingCardSection = (subject: string) => {
...
 
-  if (!defaultSubdomainName) {
-    defaultSubdomainName = "app";
-  }
   const buttonSet = CardService.newButtonSet().addButton(button);
-  const cardSection = CardService.newCardSection().addWidget(MyWalletAddress);
-
-  // allow for switching of available subdomain
+  const cardSection = CardService.newCardSection().addWidget(
+    address ? MyWalletAddress : MyEmailAddress
```
Comment (arcinston) [inline src/pages/addMeeting.ts:70]: would prefer both MyWalletAddress and MyEmailAddress widget be rendered if both exist

**Review by arcinston — CHANGES_REQUESTED** [review]

**Review by arcinston — CHANGES_REQUESTED** [review]
another thing which can be added is onGmailCompose 

can figure it out together on what the best approach would be

**Comment by arcinston** [comment]
also if i could try the test deployment on my work email : arush@huddle01.com 
would be great @ponikar

File: src/pages/addMeeting.ts (hunk 3 above)
Comment (ponikar) [reply src/pages/addMeeting.ts:70]: okay

**Comment by ponikar** [comment]
@arcinston Yes, We can setup biome later, its not a priority now, once we make sure the calcom and gcal flow is working as expected.

File: appsscript.json [hunk 4]
```diff
@@ -70,10 +71,10 @@
   "urlFetchWhitelist": [
     "https://wpss2zlpb9.execute-api.us-east-1.amazonaws.com/new-meeting",
     "https://us-central1-nfts-apis.cloudfunctions.net/createroom",
-    "https://iriko.testing.huddle01.com/verify",
     "https://gauth.huddle01.com/auth",
-    "https://api.huddle01.com/api/v1/admin/create-meeting",
-    "https://api.huddle01.com/api/v1/admin/get-subdomain"
+    "https://platform-api-darshan.huddle01.workers.dev/api/v2/calendar/subdomains",
```
[the remark this answers is not in the document]
Comment (ponikar) [reply appsscript.json:75]: Yes, we can update this once we assure its working fine.

File: src/pages/addMeeting.ts [hunk 5]
```diff
@@ -97,19 +100,20 @@ const createAddMeetingCardSection = (subject: string) => {
...
-        // ignore this entry
-        return;
-      }
-      if (subdomain.id != defaultSubdomainId) {
-        subdomainSwitch.addItem(subdomain.name, subdomain.id, false);
-      } else {
-        subdomainSwitch.addItem(subdomain.name, subdomain.id, true);
+      .setFieldName("subdomain");
+
+    subdomainResponse.subdomains.push({
```
Comment (arcinston) [inline src/pages/addMeeting.ts:105]: rm this , the api endpoint should return it by default

**Review by arcinston — CHANGES_REQUESTED** [review]

**Comment by arcinston** [comment]
@ponikar i think biome should be added , we will find some minor mistakes 
we can add it seperately in a different PR later

**Comment by arcinston** [comment]
found 2 bugs 
in case of app, it should be huddle01.app/room

image

second small issue
it says app.huddle01.com -> when it should be huddle01.app
in case of subdomain it should say subdomain.huddle01.app
 
image

**Review by arcinston — CHANGES_REQUESTED** [review]

**Pull request merged**